pkg/webhooks/pod_webhook.go

```diff
@@ -31,16 +31,23 @@ import (
 	"sigs.k8s.io/lws/pkg/utils"
 	acceleratorutils "sigs.k8s.io/lws/pkg/utils/accelerators"
 	podutils "sigs.k8s.io/lws/pkg/utils/pod"
+	"sigs.k8s.io/lws/pkg/utils/schedulerprovider"
 	statefulsetutils "sigs.k8s.io/lws/pkg/utils/statefulset"
 )
 
-type PodWebhook struct{}
+type PodWebhook struct {
+	SchedulerProvider schedulerprovider.SchedulerProvider
+}
+
+func NewPodWebhook() *PodWebhook {
+	return &PodWebhook{}
+}
 
-func SetupPodWebhook(mgr ctrl.Manager) error {
+func (p *PodWebhook) SetupPodWebhook(mgr ctrl.Manager) error {
```

Comment:
I understand that the reason behind this change. But curious about is this the conventional way of dealing with this?.

Reply:
The reason is this: Actually, I didn't change this method initially. Instead, I initialized the `SchedulerProvider` by passing a `Configuration` parameter. Since I need to add gang scheduling integration tests, but I don't want to modify the old use cases and avoid affecting them, I expected to write a `BeforeAll` in my test case so that the `SchedulerProvider` can be set up after the `PodWebhook` is initialized: https://github.com/JesseStutler/lws/blob/4889aec939fbd182580cf2d92848ad70645cea46/test/integration/webhooks/pod_test.go#L870-L872. So, I want to add a `NewPodWebhook` method and also changed the `SetupPodWebhook` to setup mgr after webhook initialized

Reply:
> I'm ok with this, just different way to initialize the webhook, can we rename to Setup then?

OK I have modified it, please check